### PR TITLE
fix : Handled the async visuals while switching themes in sub-categories pages (#2294)

### LINKED ISSUE #2294: [BUG] Light and dark theme switching.
### Description

The switching between light and dark themes works fine on the home page. However, as soon as we navigate through sub-categories and try to switch the theme, the body theme takes more time to load than its div, resulting in asynchronous visuals. 

https://github.com/rupali-codes/LinksHub/assets/139853292/5d1dfba9-e436-4ddb-ae62-ac89b7efc93f
the sub-categories.mp4 demonstrates the asynchronous switching.

### Screenshots

_No response_

### Any additional information?

_No response_

### What browser are you seeing the problem on?

Chrome, Microsoft Edge

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by Sahil-106**
## Fixes Issue

Closes #2294 

## Changes proposed

By setting the duration property to 0, I have removed the transition effect altogether. This means that when switching between themes, there won't be any animation delay in the body of sub-categories pages in applying the theme change. It will happen instantly.

## Screenshots

https://github.com/rupali-codes/LinksHub/assets/139853292/dd30b500-cf4d-44c8-bdaf-31ee48de8f7f

## Note to reviewers

As a newcomer to open source, I am excited to be part of such a vibrant community and to have the opportunity to contribute to this project. I have carefully reviewed the changes and tested them to ensure they work as intended. However, as this is my first contribution, I welcome any feedback or suggestions you may have to improve the code. thank you!

### CODE REVIEW

**Review by Anmol-Baranwal — DISMISSED** [review]
🚀 Looks good to me.

**Review by CBID2 — COMMENTED** [review]
Looks great! :) Your turn @rupali-codes

**Review by CBID2 — APPROVED** [review]
LGTM

File: layouts/GeneralLayout.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const GeneralLayout = ({ children }: { children: ReactNode }) => {
           className={clsx(
             'flex flex-col min-h-full px-4 pb-2 md:ml-[290px] lg:w-[calc(100%-290px)]',
 
-            pathname != '/' && ' transition-color ease-in-out duration-200',
+            pathname != '/' && ' transition-color ease-in-out duration-0',
```
Comment (aftabrehan) [inline layouts/GeneralLayout.tsx:36]: If we don't need the transition, please remove these classes.

```suggestion
```

**Review by aftabrehan — CHANGES_REQUESTED** [review]

File: layouts/GeneralLayout.tsx (hunk 1 above)
Comment (Sahil-106) [reply layouts/GeneralLayout.tsx:36]: Thank you for your suggestion, @aftabrehan. I will get it done.

**Comment by Sahil-106** [comment]
As per the suggestion, I have removed transition classes, and all things are working fine.

**Review by rupali-codes — COMMENTED** [review]
@Sahil-106 it looks great, but can we also add some smooth transition here?

**Comment by Sahil-106** [comment]
> @Sahil-106 it looks great, but can we also add some smooth transition here?

Yaa, this PR is all about the transition bug. Now I will create another issue for the addition of a smooth transition.

File: layouts/GeneralLayout.tsx [hunk 2]
```diff
@@ -33,8 +33,7 @@ const GeneralLayout = ({ children }: { children: ReactNode }) => {
           className={clsx(
             'flex flex-col min-h-full px-4 pb-2 md:ml-[290px] lg:w-[calc(100%-290px)]',
 
-            pathname != '/' && ' transition-color ease-in-out duration-200',
-            'bg-theme-primary-light dark:bg-inherit'
+            pathname != '/' && 'bg-theme-primary-light dark:bg-inherit'
           )}
```
Comment (aftabrehan) [inline layouts/GeneralLayout.tsx:37]: We should also remove the check `pathname != '/' &&`.

```suggestion
          className="flex flex-col min-h-full px-4 pb-2 md:ml-[290px] lg:w-[calc(100%-290px)] bg-theme-primary-light dark:bg-inherit"
```

Comment (Sahil-106) [reply layouts/GeneralLayout.tsx]: done the suggested changes.

**Review by aftabrehan — APPROVED** [review]
Excellent PR, @Sahil-106 💪🚀💯

**Comment by aftabrehan** [comment]
@rupali-codes, can you please help to review and merge the PR?

**Review by rupali-codes — APPROVED** [review]
looks perfect to me!

**Review by Anmol-Baranwal — APPROVED** [review]
🚀 Looks good to me.

**Comment by Anmol-Baranwal** [comment]
@rupali-codes 
Please merge this!

**Pull request merged**